Add bun to the Node docs (#572)
railwayapp/nixpacks: docs/pages/docs/providers/node.md

## Install

> All dependencies found in `packages.json` are installed with either NPM, Yarn, or PNPM. If Bun is detected (due to a `bun.lockb` existing), Nixpacks will install the packages with Bun.

Reviewer: ```suggestion
All dependencies found in `packages.json` are installed with either NPM, Yarn, PNPM, or Bun (depending on lockfile detected).
```
